### PR TITLE
Standardized material composition, added solvents.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Grinding reagents no longer uses the `chemical_makeup` list for results, it just converts from sheet to reagent.
- Breathing reagents no longer uses `chemical_makeup`, it will just add units of that reagent to the bloodstream directly.
- Materials now have a general `solvent_power` value. This determines if they can remove ink from paper and which other materials will dissolve in them.
- Materials that previously had `chemical_makeup` now have `dissolves_into`. This is the product list used when the reagent is mixed with a strong enough solvent (acetone, ethanol, acid).
- Heating and chilling product lists now function as ratio lists like `chemical_makeup` (and now `dissolves_into`).
- Acid melting behavior is now at material level and is governed by `solvent_power` and some adjacent vars.

### CODE REVIEW

File: code/modules/codex/categories/category_materials.dm [hunk 1]
```diff
@@ -69,12 +69,23 @@
...
+				if(mat.dissolves_in <= MAT_SOLVENT_NONE)
+					solvent_needed = "any liquid"
+				else if(mat.dissolves_in <= MAT_SOLVENT_MILD)
+					solvent_needed = "a mild solvent, like water"
+				else if(mat.dissolves_in <= MAT_SOLVENT_MODERATE)
+					solvent_needed = "a moderately strong solvent, like acetone"
+				else if(mat.dissolves_in <= MAT_SOLVENT_STRONG)
+					solvent_needed = "a strong solvent, like sulphuric acid"
+
+				material_info += "The following chemicals can be extracted from it with [solvent_needed] solvent:<br> [english_list(chems)]"
```
Comment (comma) [inline code/modules/codex/categories/category_materials.dm:97]: add solvent properties of the chem itself too pls

File: code/modules/materials/_materials.dm [hunk 2]
```diff
@@ -179,12 +178,19 @@
...
+
 	var/glass_icon = DRINK_ICON_DEFAULT
 	var/glass_name = "something"
 	var/glass_desc = "It's a glass of... what, exactly?"
 	var/list/glass_special = null // null equivalent to list()
 
 	// Matter state data.
+	var/dissolve_message = "dissolves in"
+	var/dissolve_sound = 'sound/effects/bubbles.ogg'
+	var/list/dissolves_in = MAT_SOLVENT_STRONG
```
Comment (Techhead0) [inline code/modules/materials/_materials.dm]: Should not be `/list/`

Comment (MistakeNot4892) [reply code/modules/materials/_materials.dm]: Whoops

File: code/modules/codex/categories/category_materials.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/codex/categories/category_materials.dm:97]: Done.

**Review by afterthought2 — DISMISSED** [review]

**Review by afterthought2 — DISMISSED** [review]

**Review by afterthought2 — APPROVED** [review]

**Pull request merged**